v3/sqlite/sqlite.go

```diff
@@ -0,0 +1,256 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"errors"
+	"fmt"
+	"io"
+
+	spindb "github.com/spinframework/spin-go-sdk/v3/internal/db"
+	"github.com/spinframework/spin-go-sdk/v3/internal/fermyon/spin/v2.0.0/sqlite"
+	"go.bytecodealliance.org/cm"
+)
+
+// Open returns a new connection to the database.
+func Open(name string) *sql.DB {
+	return sql.OpenDB(&connector{name: name})
+}
+
+// conn represents a database connection.
+type conn struct {
+	spinConn sqlite.Connection
+}
+
+// Close the connection.
+func (c *conn) Close() error {
+	return nil
+}
+
+// Prepare returns a prepared statement, bound to this connection.
+func (c *conn) Prepare(query string) (driver.Stmt, error) {
+	return &stmt{conn: c, query: query}, nil
+}
+
+// Begin isn't supported.
+func (c *conn) Begin() (driver.Tx, error) {
+	return nil, errors.New("transactions are unsupported by this driver")
+}
+
+// connector implements driver.Connector.
+type connector struct {
+	conn *conn
+	name string
+}
+
+// Connect returns a connection to the database.
+func (d *connector) Connect(_ context.Context) (driver.Conn, error) {
+	if d.conn != nil {
+		return d.conn, nil
+	}
+	return d.Open(d.name)
+}
+
+// Driver returns the underlying Driver of the Connector.
+func (d *connector) Driver() driver.Driver {
+	return d
+}
+
+// Open returns a new connection to the database.
+func (d *connector) Open(name string) (driver.Conn, error) {
+	results := sqlite.ConnectionOpen(name)
+	if results.IsErr() {
+		return nil, toError(results.Err())
+	}
+	d.conn = &conn{
+		spinConn: *results.OK(),
+	}
+	return d.conn, nil
+}
+
+// Close closes the connection to the database.
+func (d *connector) Close() error {
+	if d.conn != nil {
+		d.conn.Close()
+	}
+	return nil
+}
+
+type rows struct {
+	columns []string
+	pos     int
+	len     int
```

Comment:
Maybe rename to indicate this is the row count (because columns also has a length)